sql/core/src/test/scala/org/apache/spark/sql/ColumnExpressionSuite.scala

```diff
@@ -2375,4 +2376,16 @@ class ColumnExpressionSuite extends QueryTest with SharedSparkSession {
     assert(e2.getCause.isInstanceOf[RuntimeException])
     assert(e2.getCause.getMessage == "hello")
   }
+
+  test("SPARK-34677: negate/add/subtract year-month and day-time intervals") {
+    import testImplicits._
+    val df = Seq((Period.ofMonths(10), Duration.ofDays(10), Period.ofMonths(1), Duration.ofDays(1)))
+      .toDF("year-month-A", "day-time-A", "year-month-B", "day-time-B")
+    val negatedDF = df.select(-$"year-month-A", -$"day-time-A")
+    checkAnswer(negatedDF, Row(Period.ofMonths(-10), Duration.ofDays(-10)))
+    val sumDF = df.select($"year-month-A" + $"year-month-B", $"day-time-A" + $"day-time-B")
```

Comment:
`sumDF` -> `addDF`

Reply:
does `sum` function accept the new interval types?

Reply:
> does sum function accept the new interval types?

`sum` operates over `NumericType` only, see https://github.com/apache/spark/blob/3a299aa6480ac22501512cd0310d31a441d7dfdc/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Sum.scala#L49 . If you think we should support new intervals in it, let's do that separately since this PR is about operators `+`/`-`.

Reply:
I opened the JIRA SPARK-34716 to don't forget about it.